homeassistant/components/progettihwsw/binary_sensor.py

```diff
@@ -38,11 +42,12 @@ async def async_update_data():
     )
     await coordinator.async_refresh()
 
-    for i in range(1, int(input_count) + 1):
+    for i in inputs:
         binary_sensors.append(
             ProgettihwswBinarySensor(
                 coordinator,
                 f"Input #{i}",
+                board_api.create_unique_id(i, "input"),
```

Comment:
Looking upstream, these unique IDs are created based on identifiers we don't allow unique IDs to be based on.

```
    def create_unique_id(self, number, io_type: str):
        """Generate an id based on IP address and a number."""
        unencoded_ascii = (f"{self.ip}_{io_type}_{number}_{random.random()}").encode('ascii')
        base64_bytes = base64.b64encode(unencoded_ascii)
        
        return base64_bytes.decode('ascii')
```

For example: IP and random numbers. Those are not stable identifiers.

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#unique-id-requirements>


Additionally, please don't abstract unique IDs generation. They should be transparent and composed in Home Assistant.